### PR TITLE
interfaces: add login-session-observe for who, {fail,last}log and loginctl

### PR DESCRIPTION
**Pull request by jdstrand**
login-session-observe supports:
- the who, lastlog and faillog commands
- reading wtmp, utmp, lastlog and failog
- loginctl for:
  - list-sessions
  - show-session N
  - list-users
  - show-user N
  - list-seats
  - show-seat N

loginctl {seat,session,user}-status is not supported at this time since
additional accesses not strictly within the scope of this interface are
required (though due to the above accesses, they are partially
supported, but will generate apparmor denials).

### CODE REVIEW

File: interfaces/builtin/login_session_observe.go [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+const loginSessionObserveSummary = `allows observing login and session information`
```
Comment (pedronis) [inline interfaces/builtin/login_session_observe.go]: nitpick: but we tend to use "reading" not "observing" even in these -observe interface summaries

Comment (zyga) [inline interfaces/builtin/login_session_observe.go]:
```suggestion
const loginSessionObserveSummary = `allows reading login and session information`
```

File: interfaces/builtin/login_session_observe.go [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+/{,usr/}bin/lastlog ixr,
+/var/log/lastlog rk,
+
+/{,usr/}bin/faillog ixr,
+/var/log/faillog rk,
+
+# systemd session information (session files, but not .ref files)
+/run/systemd/sessions/ r,
+/run/systemd/sessions/*[0-9] rk,
```
Comment (zyga) [inline interfaces/builtin/login_session_observe.go]: Neat trick that doesn't match `.ref` files.

File: interfaces/builtin/login_session_observe.go [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+dbus (send)
+    bus=system
+    path=/org/freedesktop/login1
+    interface=org.freedesktop.login1.Manager
+    member=Get{Seat,Session,User},
+`
+
+type loginSessionObserveInterface struct {
+	commonInterface
+	secCompSnippet string
```
Comment (zyga) [inline interfaces/builtin/login_session_observe.go]: Is this meant to be extended or used later? Nothing is using it here.

Comment (jdstrand) [reply interfaces/builtin/login_session_observe.go]: No, it was a holdover from using something else as a template. Removed.

File: interfaces/builtin/login_session_observe.go (hunk 1 above)
Comment (jdstrand) [reply interfaces/builtin/login_session_observe.go]: Done

**Pull request merged**